src/Sarif.Driver/Sdk/SarifLogger.cs

```diff
@@ -102,7 +102,7 @@ public static ToolInfo CreateDefaultToolInfo(string prereleaseInfo = null)
                             }
                             });
                         }
-                        runInfo.AnalysisTargets.Add(fileReference);
+                        runInfo.FileInfo.Add(fileReference.Uri.OriginalString, new FileReference[] { fileReference });
```

Comment:
Array use in object model should be replaced with IList<T> (because arrays are of fixed size). Otherwise, producers need to have all relevant metadata in hand when populating a log file (in order to allocate array of appropriate size) or take on the inefficiency of creating a new array anytime an item is added.